packages/taquito/src/contract/big-map.ts

```diff
@@ -1,12 +1,22 @@
 import { Schema } from '@taquito/michelson-encoder';
 import BigNumber from 'bignumber.js';
 import { ContractProvider } from './interface';
+import { HttpResponseError, STATUS_CODE } from '@taquito/http-utils';
 
 export class BigMapAbstraction {
   constructor(private id: BigNumber, private schema: Schema, private provider: ContractProvider) {}
 
   async get(keyToEncode: string) {
-    return this.provider.getBigMapKeyByID(this.id.toString(), keyToEncode, this.schema);
+    try {
+      const id = await this.provider.getBigMapKeyByID(this.id.toString(), keyToEncode, this.schema);
+      return id;
+    } catch (e) {
+      if (e instanceof HttpResponseError && e.status === STATUS_CODE.NOT_FOUND) {
+        return undefined;
+      } else {
+        throw new Error(e);
```

Comment:
What do you think about just doing `throw e` instead of `throw new Error(e)` so that do not change the Error type?

Reply:
I think that's a great idea, I've changed the code and pushing up now.